test/unit/eval_algorithms/test_context_quality.py

```diff
@@ -0,0 +1,138 @@
+from typing import NamedTuple, List, Tuple
+from unittest.mock import Mock, patch
+
+import pytest
+import ray
+from ray.data import Dataset
+
+from fmeval.constants import (
+    DatasetColumns,
+    MIME_TYPE_JSONLINES,
+)
+from fmeval.data_loaders.data_config import DataConfig
+from fmeval.eval_algorithms import EvalScore, EvalOutput
+
+from fmeval.eval_algorithms.context_quality import ContextQuality, ContextQualityConfig, CONTEXT_PRECISION_SCORE
+
+
+class TestCaseContextQualityEvaluateSample(NamedTuple):
+    model_input: str
+    target_output: str
+    retrieved_context: str
+    mocked_judge_model_responses: List[Tuple]
+    expected_response: List[EvalScore]
+
+
+class TestCaseContextQualityEvaluate(NamedTuple):
+    dataset: Dataset
+    data_config: DataConfig
+    mocked_judge_model_responses: List[Tuple]
+    expected_response: List[EvalOutput]
+
+
+CONTEXT_QUALITY_DATASET = ray.data.from_items(
+    [
+        {
+            DatasetColumns.MODEL_INPUT.value.name: "What is the tallest mountain in the world?",
+            DatasetColumns.TARGET_OUTPUT.value.name: "Mount Everest.",
+            DatasetColumns.RETRIEVED_CONTEXT.value.name: "The Andes is the longest continental mountain range in the world.",
+        },
+        {
+            DatasetColumns.MODEL_INPUT.value.name: "Who won the most super bowls?",
+            DatasetColumns.TARGET_OUTPUT.value.name: "Pittsburgh Steelers<OR>New England Patriots",
+            DatasetColumns.RETRIEVED_CONTEXT.value.name: "The AFC's Pittsburgh Steelers and New England Patriots have the most Super Bowl titles at six each.",
+        },
+        {
+            DatasetColumns.MODEL_INPUT.value.name: "What do the 3 dots mean in math?",
+            DatasetColumns.TARGET_OUTPUT.value.name: "therefore sign",
+            DatasetColumns.RETRIEVED_CONTEXT.value.name: "The therefore sign is generally used before a logical consequence. <AND>The symbol consists of three dots placed in an upright triangle and is read therefore.",
+        },
+    ]
+)
+
+
+class TestContextQuality:
+    @pytest.mark.parametrize(
+        "test_case",
+        [
+            TestCaseContextQualityEvaluateSample(
+                model_input="What is the tallest mountain in the world?",
+                target_output="Mount Everest.",
+                retrieved_context="The Andes is the longest continental mountain range in the world<AND> located in South America. <AND>It stretches across seven countries and features many of the highest peaks in the Western Hemisphere. <AND>The range is known for its diverse ecosystems, including the high-altitude Andean Plateau and the Amazon rainforest.",
+                mocked_judge_model_responses=[("The score is 0", None), ("0", None), ("1", None), ("unknown", None)],
+                expected_response=[EvalScore(name=CONTEXT_PRECISION_SCORE, value=0.3333333333)],
+            )
+        ],
+    )
+    def test_context_quality_evaluate_sample(self, test_case):
```

Comment:
Can you please add more test cases for different combinations of `target_output` and `retrieved_context` length? By length, I mean the number of delimiters that are present in the string. I'd like to have more coverage for the various different cases that are possible, especially making sure to test that we `break` appropriately when handling multiple target outputs.

For both variables, you can have either length 1 or length > 1, so there's 4 combinations in that regard. Also make sure to handle the cases where you `break` early when iterating through target outputs.